### PR TITLE
Add tests for PIs and fix sampling in two dimensions

### PR DESCRIPTION
**Pull request by wreise**
#### What does this implement/fix? Explain your changes.
Test PIs. I do not manage to have a null PI for all inputs on the diagonal.

#### Any other comments?

### CODE REVIEW

File: gtda/diagrams/preprocessing.py [hunk 1]
```diff
@@ -219,7 +219,8 @@ def fit(self, X, y=None):
         if self.metric in ['landscape', 'heat', 'betti', 'persistent_image']:
```
Comment (gtauzin) [inline gtda/diagrams/preprocessing.py]: Move all the conditions on the metric inside the _discretize function (for all transformer using metrics)

**Review by gtauzin — CHANGES_REQUESTED** [review]
I just made one comment.

I think it would be better to use ``n_bins`` instead of ``n_values`` and ``_bin`` instead of ``_discretize`` but that's up to @ulupo

File: gtda/diagrams/_metrics.py [hunk 2]
```diff
@@ -63,14 +63,15 @@ def persistent_images(diagrams, sampling, step_size, weights, sigma):
...
     # Set the values outside of the sampling range to the sampling range.
-    sampled_diags[diagrams < sampling[0]] = sampling[0]
-    sampled_diags[diagrams > sampling[-1]] = sampling[-1]
-    # Birth into pixels
-    sampled_diags[:, :, 0] = np.array(
-        (sampled_diags[:, :, 0] - sampling[0]) / step_size, dtype=int)
-    # Persistence into pixels
-    sampled_diags[:, :, 1] = np.array(
-        (sampled_diags[:, :, 1] - sampling[0]) / step_size, dtype=int)
+    for k in range(2):
```
Comment (gtauzin) [inline gtda/diagrams/_metrics.py]: Call it axis instead of k and use [0, 1] instead of range.

**Review by gtauzin — CHANGES_REQUESTED** [review]

**Comment by gtauzin** [comment]
I think I know why you are experiencing this with points on the diagonal.

Make the changes and then I will merge and take over.

**Comment by wreise** [comment]
> I just made one comment.
> 
> I think it would be better to use `n_bins` instead of `n_values` and `_bin` instead of `_discretize` but that's up to @ulupo

@gtauzin Ok for the name change. The parameter name change is quite significant: the feature Transformers have the `n_values` parameter, which is used in many places (including use_cases). Afaiac, if you think it's a good idea, I will make sure that everything works.

**Review by gtauzin — APPROVED** [review]

**Pull request merged**